File: OptionalDependencies/src/main/java/com/novoda/demo/optionaldependencies/MainActivity.java
package com.novoda.demo.optionaldependencies;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        OptionalDependencies optionalDependencies = ((OptionalDependencyApplication) getApplication()).getOptionalDependencies();

        TextView paidContent = (TextView) findViewById(R.id.paid_content);
        optionalDependencies.updatePaidContent(paidContent);
    }
}

File: FirebaseNotifications/app/src/main/java/com/novoda/demo/firebasenotifications/MainActivity.kt
package com.novoda.demo.firebasenotifications

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.google.android.gms.common.ConnectionResult
import com.google.android.gms.common.GoogleApiAvailability

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (googlePlayServicesNotAvailable()) {
            makeGooglePlayServicesAvailable()
        }
    }

    private fun googlePlayServicesNotAvailable(): Boolean {
        val result = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(this)
        return result != ConnectionResult.SUCCESS
    }

    private fun makeGooglePlayServicesAvailable() {
        GoogleApiAvailability.getInstance().makeGooglePlayServicesAvailable(this)
    }
}
